client.go

```diff
@@ -29,8 +29,12 @@ const normalQueueSize = 100
 // How often to inspect the queue length.
 const reportPeriod = int64(1000)
 
-// Labelset represents a set of metric labels.
-type LabelSet map[string]string
+// LabelSet represents a set of metric labels.
+// Currently the supported value types are:
+//    string - the label value will be indexed in the metric database
```

Comment:
This seems to encourage the default to be indexed; but if indexed labels are really a precious resource, then indexing should perhaps be opt-in rather than opt-out. Eventually I envision forcing all values in a LabelSet to be explicitly marked indexed or unindexed. But I guess we can execute that transition gradually over several CLs, and I agree that for the time being using string to mean indexed is the right choice.

The only decision we're making now that we have to live with for a long time is the naming of the UnindexedLabel type. Calling the value UnindexedLabel is a little confusing to me, because the label is the key-value pair, whereas this type just denotes the value; maybe just call it UnindexedValue, or even Unindexed, so that use sites would look like

```go
labels := metricd.LabelSet{
	"foo": metricd.Unindexed("bar"),
}
```